### PR TITLE
Update Microsoft.NET.Publish.targets

### PR DESCRIPTION
**Pull request by tuespetre**
Resolves https://github.com/aspnet/websdk/issues/114

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Publish.targets [hunk 1]
```diff
@@ -56,6 +56,11 @@ Copyright (c) .NET Foundation. All rights reserved.
     </PropertyGroup>
 
     <MakeDir Directories="$(PublishDir)" />
+    
+    <ItemGroup Condition=" '$(EnableDefaultItems)' == 'true' And '$(EnableDefaultContentItems)' == 'true' ">
+      <!-- Include any files from wwwroot that may not have existed before a prepublish target -->
+      <Content Include="wwwroot/**" CopyToPublishDirectory="PreserveNewest" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(Content)" />
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Publish.targets]: Something like this is specific to the Web SDK and should go there, not in the generic .NET SDK.

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Publish.targets [hunk 2]
```diff
@@ -337,12 +342,21 @@ Copyright (c) .NET Foundation. All rights reserved.
...
+    </ItemGroup>
+
+    <AssignTargetPath Files="@(_ContentItemsToPublish)" RootFolder="$(MSBuildProjectDirectory)">
+      <Output TaskParameter="AssignedFiles" ItemName="_ContentItemsToPublishWithTargetPath" />
+    </AssignTargetPath>
+    
+    <ItemGroup>
       <_SourceItemsToCopyToPublishDirectoryAlways KeepMetadata="$(_GCTPDIKeepMetadata)"
-                                                  Include="@(ContentWithTargetPath->'%(FullPath)')"
-                                                  Condition="'%(ContentWithTargetPath.CopyToPublishDirectory)'=='Always'"/>
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Publish.targets]: It looks like `ContentWithTargetPath` is used to resolve items that should be published from referenced projects, and hence this change is breaking that and some tests that cover this are failing.

**Review by dsplaisted — CHANGES_REQUESTED** [review]
I think this should go in the Web SDK, and the way it should be injected is probably closer to the workaround described [here](https://github.com/aspnet/websdk/issues/114#issuecomment-280076533), where the new items are added to `ContentWithTargetPath`.

File: src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Publish.targets (hunk 1 above)
Comment (tuespetre) [inline src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Publish.targets]: Shoot, you are right. I was mixed up with the DefaultItems.props file.

**Comment by tuespetre** [comment]
Will redo this in the web SDK. Thanks 🌞

**Comment by dsplaisted** [comment]
@tuespetre Thanks a bunch, please tag me when you send the PR to the web SDK so I see it.

**Pull request closed without merging**